src/JJMasterData.Core/WebComponents/JJDownloadFile.cs

```diff
@@ -96,26 +93,35 @@ private HtmlBuilder GetDownloadHtmlElement()
         return html;
     }
     
-    internal void ResponseDirectDownload()
+    internal void DirectDownload()
     {
         if (string.IsNullOrEmpty(FilePath))
             throw new ArgumentNullException(nameof(FilePath));
 
-        var fileName = FilePath.Substring(FilePath.LastIndexOf('\\') + 1);
         if (!File.Exists(FilePath))
         {
             Log.AddError(Translate.Key("File not found at {0}", FilePath));
-            throw new Exception(Translate.Key("File {0} not found!", fileName));
+            throw new Exception(Translate.Key("File {0} not found!", FilePath));
         }
 
-        CurrentContext.Response.ClearResponse();
-        CurrentContext.Response.AddResponseHeader("Content-Type", MimeTypeUtil.GetMimeType(fileName));
-        CurrentContext.Response.AddResponseHeader("Content-Transfer-Encoding", "binary");
-        CurrentContext.Response.AddResponseHeader("Content-Description", "file Transfer");
-        CurrentContext.Response.AddResponseHeader("Content-Disposition",
-            $"attachment; filename={HttpUtility.UrlEncode(fileName)}");
+        DirectDownload(FilePath);
```

Comment:
This solves the problem bellow on YARP systems.
![image](https://user-images.githubusercontent.com/52143624/207384644-88880774-7a98-4203-a120-15807bb688c6.png)